--- a/net/minecraft/server/network/ConfigurationTask.java
+++ b/net/minecraft/server/network/ConfigurationTask.java
@@ -2,6 +_,7 @@
 
 import java.util.function.Consumer;
 import net.minecraft.network.protocol.Packet;
+import net.minecraft.resources.ResourceLocation;
 
 public interface ConfigurationTask {
    void start(Consumer<Packet<?>> p_294184_);
@@ -9,6 +_,10 @@
    ConfigurationTask.Type type();
 
    public static record Type(String id) {
+      public Type(ResourceLocation location) {
+            this(location.toString());
+      }
+      
       @Override
       public String toString() {
          return this.id;
