--- a/net/minecraft/network/protocol/login/ClientboundCustomQueryPacket.java
+++ b/net/minecraft/network/protocol/login/ClientboundCustomQueryPacket.java
@@ -2,6 +_,7 @@
 
 import net.minecraft.network.FriendlyByteBuf;
 import net.minecraft.network.protocol.Packet;
+import net.minecraft.network.protocol.common.custom.DiscardedPayload;
 import net.minecraft.network.protocol.login.custom.CustomQueryPayload;
 import net.minecraft.network.protocol.login.custom.DiscardedQueryPayload;
 import net.minecraft.resources.ResourceLocation;
@@ -12,11 +_,11 @@
    public ClientboundCustomQueryPacket(FriendlyByteBuf p_179810_) {
       this(p_179810_.readVarInt(), readPayload(p_179810_.readResourceLocation(), p_179810_));
    }
-
+   
    private static CustomQueryPayload readPayload(ResourceLocation p_295267_, FriendlyByteBuf p_295117_) {
       return readUnknownPayload(p_295267_, p_295117_);
    }
-
+   
    private static DiscardedQueryPayload readUnknownPayload(ResourceLocation p_294837_, FriendlyByteBuf p_296380_) {
       int i = p_296380_.readableBytes();
       if (i >= 0 && i <= 1048576) {
